crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -535,6 +537,83 @@ def cancel
         upgrade_status.initialize_state
       end
 
+      def upgrade_nodes_disruptive(to_upgrade)
```

Comment:
Please add a comment that this is really working with elements and not nodes. (Maybe even rename it).

Also I suggest to rename ```to_upgrade``` to ```elements_to_upgrade```